ApiGateway/src/main/java/app/controllers/StrategyController.java

```diff
@@ -79,7 +80,9 @@ public Strategy updateStrategy(@PathVariable("id") Integer id, @RequestBody Stra
 
     @GetMapping("/strategies/{id}/trades")
     public List<Trade> getTradesByStrategyId(@PathVariable("id") Integer id){
-        return strategyService.getTradesByStrategyId(id);
+        List<Trade> trades = strategyService.getTradesByStrategyId(id);
+        trades.sort(Comparator.comparing(Trade::getTimeTransacted));

```

Comment:
Given that the program is asynchronous. For Bollingers, the trades were sometimes out of order when returned to the front-end. This is simply guaranteeing that the trades returned will always be in time stamp order.